website/server/controllers/api-v3/tasks.js

```diff
@@ -678,6 +678,10 @@ api.updateTask = {
       task.group.sharedCompletion = sanitizedObj.sharedCompletion;
     }
 
+    if (task.frequency === 'monthly' && task.startDate) {
```

Comment:
thanks @laurel-thomson ! The PR works well but I have one doubt, pinging @Alys for an extra check.

When a daily is set to repeat monthly `task.frequency === 'monthly'` it can repeat on the same day every month (for example the 18th of every month) or on `Day of the week` (as can be chosen on the website).

The difference between the two repetition schedules is defined by whether the `task.daysOfMonth` or `task.weeksOfMonth` arrays have values in them.

So from my understanding this bug only applies to `daysOfMonth` and it should not be set when `task.weeksOfMonth` has one or more values in it.

@Alys will be able to confirm but if I'm correct this code should only be applied when `task.weeksOfMonth` is empty or `task.daysOfMonth` is not before the update

Reply:
Just to let you know, I'm going to try to look into this today. Depending on work pressures though, it may have to slip a day or two. Sorry @laurel-thomson I'll try to get to it as soon as I can!

Reply:
Yes, paglias is right. The `task.daysOfMonth` array should be given a new value only if the player has specified that they want to use the "Day of the Month" "Repeat On" option.

Currently, the website front-end is responsible for setting either `task.daysOfMonth` or `task.weeksOfMonth` depending on which "Repeat On" option the user has chosen ("Day of the Month" or "Day of the Week") - see the website code linked from my  original issue https://github.com/HabitRPG/habitica-android/issues/1272.
The API doesn't yet have the ability to make that decision, and I think that moving all of that front-end code into the API would be a breaking change - i.e., not something we want to do.

Fortunately I don't think moving it all into the API will be necessary to fix the original issue. I think all that's needed is to change the value in the `task.daysOfMonth` array only if the array already has a value. So you can add a check for whether it has a value to your `if` statement on line 681.

Reply:
> The API doesn't yet have the ability to make that decision, and I think that moving all of that front-end code into the API would be a breaking change - i.e., not something we want to do.

agreed, a simple if to check the array will be enough in this case

Reply:
Sounds good!  I've updated the PR